lib/buffer.js

```diff
@@ -7,62 +7,219 @@ const alloc = smalloc.alloc;
 const truncate = smalloc.truncate;
 const sliceOnto = smalloc.sliceOnto;
 const kMaxLength = smalloc.kMaxLength;
+const useOldBuffer = !!process.useOldBuffer;
 
-exports.Buffer = Buffer;
-exports.SlowBuffer = SlowBuffer;
-exports.INSPECT_MAX_BYTES = 50;
+var Buffer, SlowBuffer, NativeBuffer;
 
 
-Buffer.poolSize = 8 * 1024;
-var poolSize, poolOffset, allocPool;
+// New Buffer implementation using Uint8Array. Done this way so the constructor
+// name can remain "Buffer" and "SlowBuffer".
 
+if (!useOldBuffer) {
```

Comment:
Suggestion: instead of having `if` statements everywhere, maybe move the old and new code to two files buffer_new.js and buffer_old.js in lib/internal?  Then you just require() the right one here and re-export it.